pkg/tracing/tracing.go

```diff
@@ -18,26 +18,154 @@
 package tracing
 
 import (
-	"io"
-	"time"
-
+	"context"
+	"encoding/json"
+	"fmt"
 	"github.com/megaease/easegress/pkg/util/fasttime"
-
-	zipkingo "github.com/openzipkin/zipkin-go"
-	zipkinreporter "github.com/openzipkin/zipkin-go/reporter"
-	zipkingohttp "github.com/openzipkin/zipkin-go/reporter/http"
+	"go.opentelemetry.io/contrib/propagators/b3"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/jaeger"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
+	"go.opentelemetry.io/otel/exporters/zipkin"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.12.0"
+	"go.opentelemetry.io/otel/trace"
+	"net"
+	"net/http"
+	"time"
 )
 
 type (
+	headerFormat string
 	// Spec describes Tracer.
 	Spec struct {
-		ServiceName string            `json:"serviceName" jsonschema:"required"`
-		Tags        map[string]string `json:"tags" jsonschema:"omitempty"`
-		Zipkin      *ZipkinSpec       `json:"zipkin" jsonschema:"required"`
+		ServiceName  string                `json:"serviceName" jsonschema:"required,minLength=1"`
+		Attributes   map[string]string     `json:"attributes" jsonschema:"omitempty"`
```

Comment:
better keep both `Tags` and `Attributes`, and ignore `Tags` if `Attributes` is not empty, this result in better backward compatibility.